cmd/worker/funcs_kmod_tarball.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/worker"
+	"github.com/spf13/cobra"
+)
+
+var kmodTarballCmd = &cobra.Command{
+	Use:               "kmod-tarball",
+	Short:             "Manage kernel modules using tarball",
+	PersistentPreRunE: kmodTarballFuncPreRunE,
+}
+
+var kmodTarballLoadCmd = &cobra.Command{
+	Use:   "load",
+	Short: "Load a kernel module using tarball file",
+	Args:  cobra.ExactArgs(1),
+	RunE:  kmodLoadFunc,
+}
+
+var kmodTarballUnloadCmd = &cobra.Command{
+	Use:   "unload",
+	Short: "Unload a kernel module",
+	Args:  cobra.ExactArgs(1),
+	RunE:  kmodUnloadFunc,
+}
+
+func kmodTarballFuncPreRunE(cmd *cobra.Command, args []string) error {
+	err := cmd.Parent().PersistentPreRunE(cmd.Parent(), args)
+	if err != nil {
+		return fmt.Errorf("failed to call root command pre-run: %v", err)
+	}
+	ip := worker.NewTarImageMounter(worker.ImagesDir, logger)
+	mr := worker.NewModprobeRunner(logger)
+	w = worker.NewWorker(ip, mr, logger)
+
+	return nil
+}
+
+func configureTarballCommands() {
+	rootCmd.AddCommand(kmodCmd, kmodTarballCmd)
+	kmodTarballCmd.AddCommand(kmodTarballLoadCmd, kmodTarballUnloadCmd)
+
+	kmodTarballLoadCmd.Flags().String(
+		worker.FlagFirmwareClassPath,
+		"",
+		"if set, this value will be written to "+worker.FirmwareClassPathLocation,
+	)
+
+	kmodTarballLoadCmd.Flags().String(
+		worker.FlagFirmwareMountPath,
+		"",
+		"if set, this the value that firmware host path is mounted to")
+
+	kmodTarballUnloadCmd.Flags().String(
+		worker.FlagFirmwareMountPath,
+		"",
+		"if set, this the value that firmware host path is mounted to")
```

Comment:
I think we should try to reuse the function calls in `main` that apply to `kmodLoadCmd` and `kmodUnloadCmd`.
If we decide to go with `--tarball` instead, we can eliminate this new code entirely.

Reply:
i can add a re-use function in case we agree on the remark above